File: CS/WizPageDataOption.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using DevExpress.XtraEditors;
using DevExpress.XtraReports.Native;
using DevExpress.XtraReports.Design;
using DevExpress.Utils;

namespace RepWizardCustomQuery
{
    public partial class WizPageDataOption : DevExpress.Utils.InteriorWizardPage
    {
        XtraReportWizardBase standardWizard;

        public WizPageDataOption(XRWizardRunnerBase runner)
        {
            this.standardWizard = runner.Wizard;

            InitializeComponent();
        }

        protected override bool OnSetActive()
        {
            Wizard.WizardButtons = WizardButton.Back | WizardButton.Next | WizardButton.DisabledFinish;
            return true;
        }

        protected override string OnWizardNext()
        {
            if (radioGroup1.SelectedIndex != 0)
            {
                return "WizPageTables";
            }
            else
            {
                return "WizPageQuery";
            }
        }
    }
}

File: VB/WizPageDataOption.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Text
Imports System.Windows.Forms
Imports DevExpress.XtraEditors
Imports DevExpress.XtraReports.Native
Imports DevExpress.XtraReports.Design
Imports DevExpress.Utils

Namespace RepWizardCustomQuery
	Partial Public Class WizPageDataOption
		Inherits DevExpress.Utils.InteriorWizardPage
		Private standardWizard As XtraReportWizardBase

		Public Sub New(ByVal runner As XRWizardRunnerBase)
			Me.standardWizard = runner.Wizard

			InitializeComponent()
		End Sub

		Protected Overrides Function OnSetActive() As Boolean
			Wizard.WizardButtons = WizardButton.Back Or WizardButton.Next Or WizardButton.DisabledFinish
			Return True
		End Function

		Protected Overrides Function OnWizardNext() As String
			If radioGroup1.SelectedIndex <> 0 Then
				Return "WizPageTables"
			Else
				Return "WizPageQuery"
			End If
		End Function
	End Class
End Namespace